src/events/SDL_pen.c

```diff
@@ -346,7 +346,7 @@ void SDL_SendPenTouch(Uint64 timestamp, SDL_PenID instance_id, SDL_Window *windo
         if (eraser && ((input_state & SDL_PEN_INPUT_ERASER_TIP) == 0)) {
             input_state |= SDL_PEN_INPUT_ERASER_TIP;
             send_event = true;
-        } else if (!down && (input_state & SDL_PEN_INPUT_ERASER_TIP)) {
+        } else if (!eraser && (input_state & SDL_PEN_INPUT_ERASER_TIP)) {
```

Comment:
I don't think this change is correct. I think eraser is just a flag to know whether the contact tool is an eraser, and the down flag tells whether it's in contact or not. @icculus, can you confirm?

Reply:
Nope, this fix is correct; it's deciding whether the pen's state has moved from the pen side to the eraser side, regardless of whether it is touching. It's a copy/paste error from the `else if` above it.

Good catch!